src/renderers/shared/stack/reconciler/ReactComponentEnvironment.js

```diff
@@ -21,16 +30,16 @@ var ReactComponentEnvironment = {
    * Optionally injectable hook for swapping out mount images in the middle of
    * the tree.
    */
-  replaceNodeWithMarkup: null,
+  replaceNodeWithMarkup: (null: ?ReplaceNodeWithMarkup),
```

Comment:
I'm not sure how to type those. In the comment it says it is optional but there's no checks in practice in the call sites: https://github.com/facebook/react/blob/a229cdba7fd00799d82c1bab23704220d47e6331/src/renderers/shared/stack/reconciler/ReactCompositeComponent.js#L1024

If I put it required, then flow won't let me type it. I'm not sure if we should update the call sites with a check or lie to flow and say it is always defined and trust the user to properly inject it.